packages/odyssey-react/src/components/TextInput/TextInput.tsx

```diff
@@ -179,12 +215,28 @@ export const TextInput = withTheme(
           }
         : {};
 
-    const prefixStyles = useCx(
-      styles.prefix,
-      isSearchTextInput && styles.affixIcon
+    const isSearchTextInput = type === "search";
+    // Prefix style and logic
+    const isPrefixIcon = isSearchTextInput;
+    const prefixIconName = "search";
+    const prefixStyles = useCx(styles.prefix, isPrefixIcon && styles.affixIcon);
+
+    // Suffix style and logic
+    const isSuffixButton = isSearchTextInput;
+    const suffixButtonIconName = "close-circle-filled";
+    const showSuffixButton = isControlled ? !!value : !!hasUncontrolledValue;
```

Comment:
why show the suffix button in a controlled component with a value? The user will not be able to remove it so it seems a bit suspect to include that UI affordance?

Reply:
We may want to address this behavior alongside the Password code to address variation. For example, a Controlled Search is unclearable, but a Controlled Click-to-Copy or Password field would still find the suffix useful.

Reply:
For the controlled case, the button still fires the onChange with an empty value. I think that is still useful